Automattic/jetpack
# Add Google Fonts provider and register a selection of fonts

#### Changes proposed in this Pull Request:

* Adds a new Jetpack package with a font provider for Google fonts, based on the work from https://github.com/WordPress/wordpress-develop/pull/1736, and using the Simplified API from https://github.com/WordPress/gutenberg/pull/37140.
* Adds a Jetpack plugin module that registers a selection of Google Fonts from the Jetpack Global Styles feature (https://github.com/Automattic/wp-calypso/blob/3fff686fba8d74690cd03ec5f10ac43ec3e046b3/apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php#L84)

#### Jetpack product discussion

paYE8P-1qo-p2

#### Does this pull request change what data or activity we track or use?

Loads Google Fonts settings and font files using Google APIs, if the module is activated.

#### Testing instructions:

Install the Gutenberg plugin with changes from this PR https://github.com/WordPress/gutenberg/pull/37140 (now in Gutenberg `trunk`, but not yet released).

* `pnpm jetpack install -r && pnpm jetpack install --all` to install the new fonts provider package.
* Activate the Jetpack plugin and activate the Google Fonts module from `/wp-admin/admin.php?page=jetpack_modules`
* Activate a block based theme, such as Twenty Twenty Two
* Go to the Site editor and open the Global Styles > Text panel and select one of the new font choices.

<img width="492" alt="image" src="https://user-images.githubusercontent.com/1699996/155218297-ef8c69ce-3200-4e12-9088-15ff18af16e7.png">

* Save your changes and check that the font loads on the front end of the Site.
* Open the Post editor and add a block that has font styles, like a button block.
* Select one of the font options, save, and check the front-end.

<img width="855" alt="image" src="https://user-images.githubusercontent.com/1699996/155218222-7257350a-ded0-4e38-b657-e698f17db240.png">
